Sources/Build/LLBuildProgressTracker.swift

```diff
@@ -180,8 +180,10 @@ final class LLBuildProgressTracker: LLBuildBuildSystemDelegate, SwiftCompilerOut
         } ?? [:]
         self.swiftParsers = swiftParsers
 
-        self.taskTracker.onTaskProgressUpdateText = { progressText, _ in
-            self.queue.async {
+        self.taskTracker.onTaskProgressUpdateText = { [weak self] progressText, _ in
+            guard let self else { return }
+            self.queue.async { [weak self] in
+              guard let self else { return }
```

Comment:
NIT: align the spacing
```suggestion
                guard let self else { return }
```